testing/test_windows_subst.py

```diff
@@ -0,0 +1,95 @@
+import os.path
+import subprocess
+import sys
+from string import ascii_lowercase
+
+from py._path.local import LocalPath
+
+import pytest
+from _pytest import pytester
+from _pytest.fixtures import FixtureRequest
+
+
+class _SubstTmpdirFactory:
+    """
+    A temp dir factory which may do a subst on Windows drives.
+    """
+
+    def __init__(self, tmpdir, tmpdir_factory, subst):
+        self.tmpdir = tmpdir
+        self.tmpdir_factory = tmpdir_factory
+        self.subst = subst
+        self.subst_drives = {}
+
+    def _subst_path(self, filename):
+
+        if self.subst:
+            for c in ascii_lowercase[7:]:  # Create a subst drive from H-Z.
+                c += ":"
+                if not os.path.exists(c):
+                    drive = c
+                    break
+            else:
+                raise AssertionError("Unable to find suitable drive letter for subst.")
+
+            directory = os.path.dirname(filename)
+            basename = os.path.basename(filename)
+
+            args = ["subst", drive, directory]
+            subprocess.check_call(args)
+            assert os.path.exists(drive)
+            self.subst_drives[drive] = directory
+
+            filename = LocalPath(os.path.join(drive, basename))
+
+        return filename
+
+    def mktemp(self, name, numbered):
+        filename = self.tmpdir_factory.mktemp(name, numbered=numbered)
+        filename = self._subst_path(filename)
+        return filename
+
+    def unsubst(self):
+        for drive in self.subst_drives:
+            args = ["subst", "/D", drive]
+            subprocess.check_call(args)
+
+
+@pytest.fixture(params=[{"subst": True}, {"subst": False}])
+def _subst_tmpdir_factory(request: FixtureRequest, tmpdir, tmpdir_factory):
```

Comment:
```suggestion
def subst_tmpdir_factory(request: FixtureRequest, tmpdir, tmpdir_factory):
```

No need to use `_` here, this fixture is already local to the module.